core/network/backend/src/main/java/com/appcoins/wallet/core/network/backend/remote/RemoteTransactionDataSource.kt

```diff
@@ -0,0 +1,15 @@
+package com.appcoins.wallet.core.network.backend.remote
+
+import com.appcoins.wallet.core.network.backend.BackendDataSource
+import com.appcoins.wallet.core.network.backend.api.TransactionsApi
+import javax.inject.Inject
+
+class RemoteTransactionDataSource @Inject constructor(private val api: TransactionsApi) :
+  BackendDataSource {
+
+  override suspend fun getTransactionsHistory(
+    wallet: String,
+    selectedCurrency: String,
+    limit: Int
+  ) = api.getTransactionHistory(wallet = wallet, defaultCurrency = selectedCurrency, limit = limit)
+}
```

Comment:
Isn’t this simply an unecessary abstraction?
The api could be called direclty in the repository instead of having this codesmell (middle man) doing only the job of calling the api anyway.
Combined with its interface, we have 2 classes plus 2 interfaces that could be simply one repository since we aren’t using the repo interface for more than one implements

Reply:
I don’t think so, suppose I need to get some data from the local database and some data from network, don’t you prefer to have that separatedly?

Reply:
I get that, but to have a class do simply call the api and return that, acheives nothing other than adding complexity to the code.
If the Repository need to get data from the network and from the database, the repository should call the network and database instead of having a class and interface just to do this ‘api.getTransactionHistory(wallet = wallet, defaultCurrency = selectedCurrency, limit = limit)’.
Calling the api directly or calling this dataSource its the same in terms of separation but this implemetantion adds complexity because it adds a middle man class and adds an interface that will always only have one implementation

Reply:
I agree with you, done!